plugins/pluginmanagers/lazy.nix

```
@@ -1,222 +1,204 @@
-              '';
-
-              opts =
-                with helpers.nixvimTypes;
-                helpers.mkNullOrOption (maybeRaw (attrsOf anything)) ''
-                  opts should be a table (will be merged with parent specs),
-                  return a table (replaces parent specs) or should change a table.
-                  The table will be passed to the Plugin.config() function.
-                  Setting this value will imply Plugin.config()
+                main = mkNullOrOption str ''
+                  You can specify the main module to use for config() and opts(),
+                  in case it can not be determined automatically. See config()
                 '';
-            };
-          });
-
-          listOfPlugins = types.listOf pluginType;
-        in
-        mkOption {
-          type = listOfPlugins;
-          default = [ ];
-          description = "List of plugins";
+
+                submodules = defaultNullOpts.mkBool true ''
+                  When false, git submodules will not be fetched.
+                  Defaults to true
+                '';
+
+                event =
+
+                  mkNullOrOption (maybeRaw (
+                    either str (listOf str)
+                  )) "Lazy-load on event. Events can be specified as BufEnter or with a pattern like BufEnter *.lua";
+
+                cmd = mkNullOrOption (maybeRaw (either str (listOf str))) "Lazy-load on command";
+
+                ft = mkNullOrOption (maybeRaw (either str (listOf str))) "Lazy-load on filetype";
+
+                keys = mkNullOrOption (maybeRaw (either str (listOf str))) "Lazy-load on key mapping";
+
+                module = mkNullOrOption (enum [ false ]) ''
+                  Do not automatically load this Lua module when it's required somewhere
+                '';
+
+                priority = mkNullOrOption number ''
+                  Only useful for start plugins (lazy=false) to force loading certain plugins first.
+                  Default priority is 50. It's recommended to set this to a high number for colorschemes.
+                '';
+
+                optional = defaultNullOpts.mkBool false ''
+                  When a spec is tagged optional, it will only be included in the final spec,
+                  when the same plugin has been specified at least once somewhere else without optional.
+                  This is mainly useful for Neovim distros, to allow setting options on plugins that may/may not be part
+                  of the user's plugins
+                '';
+
+                opts =
+
+                  mkNullOrOption (maybeRaw (attrsOf anything)) ''
+                    opts should be a table (will be merged with parent specs),
+                    return a table (replaces parent specs) or should change a table.
+                    The table will be passed to the Plugin.config() function.
+                    Setting this value will imply Plugin.config()
+                  '';
```

```suggestion
                  mkNullOrOption (maybeRaw (attrsOf anything)) ''
                    The opts value can be one of the following:

                    - A table: This table will be merged with any existing configuration settings from parent specifications.
                    - A function that returns a table: The returned table will completely replace any existing configuration settings from parent specifications.
                    - A function that modifies a table: This function will receive the existing configuration table as an argument and can modify it directly.

                     In all cases, the resulting configuration table will be passed to the Plugin.config() function.
                     Setting the opts value automatically implies that Plugin.config() will be called.
                  '';
```

I think this might make it more clear what is occurring? Maybe we should link to their docs, too?

---

That sounds like lazy.nvim is telling us this option can be used to configure plugins that **may or may not** be loaded.

It is saying that a different part of lazy's configuration is responsible for saying what/when plugins are loaded.

---

Re: `gw` and `gq`, I generally wrap manually. I usually try to wrap semantically; preferring line breaks at punctuation or the end of a phrase if possible. Unless it looks too silly and unbalanced.